src/Sentry.Protocol/BaseScopeExtensions.cs

```diff
@@ -308,38 +307,8 @@ public static void Apply(this BaseScope from, BaseScope to)
         /// <param name="state">The state object to apply.</param>
         public static void Apply(this BaseScope scope, object state)
         {
-            switch (state)
-            {
-                case string scopeString:
-                    // TODO: find unique key to support multiple single-string scopes
-                    scope.SetTag("scope", scopeString);
-                    break;
-                case IEnumerable<KeyValuePair<string, string>> keyValStringString:
-                    scope.SetTags(keyValStringString
-                        .Where(kv => !string.IsNullOrEmpty(kv.Value)));
-                    break;
-                case IEnumerable<KeyValuePair<string, object>> keyValStringObject:
-                    {
-                        scope.SetTags(keyValStringObject
-                            .Select(k => new KeyValuePair<string, string>(
-                                k.Key,
-                                k.Value?.ToString()))
-                            .Where(kv => !string.IsNullOrEmpty(kv.Value)));
-
-                        break;
-                    }
-#if HAS_VALUE_TUPLE
-                case ValueTuple<string, string> tupleStringString:
-                    if (!string.IsNullOrEmpty(tupleStringString.Item2))
-                    {
-                        scope.SetTag(tupleStringString.Item1, tupleStringString.Item2);
-                    }
-                    break;
-#endif
-                default:
-                    scope.SetExtra("state", state);
-                    break;
-            }
+            var processor = scope.ScopeOptions.SentryScopeProcessor ?? new DefaultSentryScopeProcessor();
```

Comment:
We'll allocate this processor for each call to `Apply`, what about we set this instance to the Options ctor (and make it non null), users can set a new one and replace it though

Reply:
It is already like that on `SentryOptions`, but we can't guarantee it on the interface level (`IScopeOptions`) unless we unconditionally trust that user will respect NRTs. The RHS of `??` should not be executed majority of the time.